### DISCUSSION TITLE
Content models with custom storefront on vercel, using sveltekit

### DISCUSSION THREAD

**Discussion by torrfura** [swellstores/community#264] [Help] [answered]
Hi!

I'm finding the documentation a bit hard to wrap my head around. In which end do I start if I'd like to use SvelteKit instead of Next JS for example, building my custom models / storefront, deploying on vercel?

Firstly, I can't figure out how I can utilise the visual page editor in swell dashboard, using my custom content models that I've copied from the Horizon theme. I'd love some help or guidance in the docs for getting started. Can't find any good pointers in the docs.

Secondly, is it possible to have a custom theme, pointing on a local environment while developing those content models?

**Comment by rodrigoddalmeida** [comment]
Hi @torrfura! While we're still updating docs for development flows based on Origin and Horizon (our starter themes), your case seems to be even more custom.

Our visual editor does support custom content models from Origin and Horizon even if you are using a different framework or setup, but then you need to implement event listeners in your storefront app so it can refresh as you change settings and content in the Editor. We don't fully support this workflow at the moment, that's why we don't have documentation available.

I can help you through the process though, if you choose to go that way - for starters, you can take a look at how Origin and Horizon have setup their event listeners to work with the Editor and go from there.

Origin (Vue/Nuxt.js): https://github.com/swellstores/origin-theme/blob/master/modules/swell-editor/plugin.js
Horizon (React): https://github.com/swellstores/horizon/blob/main/hooks/useLiveEditorUpdates.ts

**Comment by torrfura** [comment]
Thank you! I totally understand and will dig deeper into livereload. My question though is how do I even get started with custom content models, such as those you have in Horizon-theme for example, where I can see the /config folder containing all json configurations.

I've tried copy them over but I cannot understand how to connect them to the Swell admin interface, so it's possible to add custom pages etc?

**Reply by rodrigoddalmeida** [reply, marked as answer]
We support a strict set of Editor configs, so content models are not fully supported - only Pages and Quizzes are detected by the Editor and will render fields appropriately based on the schema set by `content/*.json`. You can add fields to Editor > Settings via `editor.json`.

You can push/pull content and config .json files via CLI: https://www.npmjs.com/package/@swell/cli
Here's a reference on how it is done in Horizon: https://github.com/swellstores/horizon/blob/main/build-utils/sync-storefront.mjs

You can find more info on how to customize some of the content and settings fields in the README for Origin and Horizon, they provide some insight on supported types and syntax.

Let me know how it goes.
